homeassistant/helpers/device_registry.py

```diff
@@ -81,7 +82,7 @@ def async_get_device(self, identifiers: set, connections: set):
     def async_get_or_create(self, *, config_entry_id, connections=None,
                             identifiers=None, manufacturer=_UNDEF,
                             model=_UNDEF, name=_UNDEF, sw_version=_UNDEF,
-                            via_hub=None):
+                            via_hub=None, area_id=None):
```

Comment:
area_id should not come from get_or_create. It can only be passed in via a direct update. We need a new method like [this](https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/helpers/entity_registry.py#L152-L160)